### ISSUE TITLE
remove_npcs causes ctd without log

### ISSUE THREAD

**Issue by RenechCDDA** [CleverRaven/Cataclysm-DDA#62369]
### Describe the bug

See #60938 for a specific occurrence of the issue. Game hangs for a moment or two and then crashes without producing a log. According to investigations by @mqrause this is caused by a combination of the compass sidebar widget trying to update when forced redraw is enabled.

Call stack:
![image](https://user-images.githubusercontent.com/84619419/203868218-f50d4b9b-ceff-4843-a352-cd7a6cc560a8.png)

### Steps to reproduce

http://files.catbox.moe/oevx9t.zip Load this save. Walk north in any build before 18761d73c027906c3ad1cbf42a719fe5eeb815f4.
(Hosted 3rd party due to filesize of ~41MB. In case of loss, another save is available in #60938)

### Expected behavior

Crashing is not desired.

### Versions and configuration

http://files.catbox.moe/oevx9t.zip Load this save. Walk north in any build before 18761d73c027906c3ad1cbf42a719fe5eeb815f4.
(Hosted 3rd party due to filesize of ~41MB. In case of loss, another save is available in #60938)

### Additional context

@I-am-Erk suggested applying debug invisibility to the NPC before mapgen removes him. I locally changed the EOC trigger to this:

```    "speaker_effect": {
      "effect": [
        { "u_add_var": "ANCHOR_SELLER_paranormal", "type": "dialogue", "context": "BEM_ANCHOR_SELLER", "value": "yes" },
	{ "npc_add_trait": "DEBUG_CLOAK" },
        { "queue_eocs": ["EOC_BEM_ANCHOR_SELLER_GONE"], "time_in_future": [ "0s", "0s" ] }
```

It prevents the player from crashing in most instances, but it notably does not work against clairvoyance, which may be acquired with artifacts.
A further suggestion was to change the delay trigger to 1s, like this:
```
        { "queue_eocs": "EOC_BEM_ANCHOR_SELLER_GONE", "time_in_future": [ "1s", "1s" ] }
```

This also prevents the crash in all circumstances that were tested. It is unclear to me or anyone involved *why*. Per @mqrause's discord participation: "[...] I've put up #62362 to fix the crash. But I couldn't figure out what causes the state (he's still visible after completing the dialog) that could lead to the crash in the first place."

**Comment by RenechCDDA** [comment]
/confirming my own issue as I was asked to post this so we could keep track of the general issue.

**Comment by I-am-Erk** [comment]
I suspect the answer to this lies in the compass rendering code, but what do I know? Not much in this case.

**Comment by andrei8l** [comment]
Can't reproduce on linux with 8201abd4c761e493c47c597b7599ec381729d997 and ASan+UBSan from the Winstead or Pine Ridge saves. Tried with all the compass widgets from the classic/labels/structured layouts. Tried with `Force Redraw` both on and off.

This might be a mscv-only issue. Any chance you can build with ASan and get a log from it?

**Comment by RenechCDDA** [comment]
> This might be a mscv-only issue. Any chance you can build with ASan and get a log from it?

I passed my IDE's(Visual Studio 2019) compiler an argument of `/fsanitize=address` which appears to be what you are referring to? I'm not usually on the forefront of this level of debugging. I also passed cataclysm-tiles.exe an argument of `set ASAN_SAVE_DUMPS=DumpyMcDump.dmp`. After the crash occurs I stepped through the breakpoints, letting it continue until the process terminated. No file by that name could be found. But here's the output window, which might be what you want?

[visual studio output during crash.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/10087967/visual.studio.output.during.crash.txt)
